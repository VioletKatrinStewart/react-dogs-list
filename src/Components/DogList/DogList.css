.pinkdiv {
  background-color: pink;
  text-align: center;
}

.doglink {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 30px;
}
