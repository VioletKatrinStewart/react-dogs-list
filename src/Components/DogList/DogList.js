import React from 'react';
import { Link } from 'react-router-dom';

export default function DogList({ dogs }) {
  console.log(dogs);
  return (
    <div>
      {dogs.map((dog) => (
        <Link key={dogs.id} to={`/dogs/${dogs.id}`}>
          {dog.name}
          <img src={dog.image} />
        </Link>
      ))}
    </div>
  );
}
