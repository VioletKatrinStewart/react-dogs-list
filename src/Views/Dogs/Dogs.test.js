import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DogList from './Dogs';

it('should render a list of dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <DogList />
    </MemoryRouter>
  );
  await screen.findByText('Stinky');
  expect(container).toMatchSnapshot();
});
