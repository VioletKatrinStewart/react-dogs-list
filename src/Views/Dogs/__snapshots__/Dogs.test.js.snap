// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of dogs 1`] = `
<div>
  <div
    class="pinkdiv"
  >
    <header>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <a
        href="/admin"
      >
        Admin
      </a>
    </header>
    <h1
      class="text"
    >
      WELCOME TO DOG ADOPTION
    </h1>
    <h2
      class="text"
    >
      These are the dogs we have available to adopt, click for more details!
    </h2>
    <div
      class="doglist"
    >
      <a
        class="doglink"
        href="/dogs/6"
      >
        <h3>
           
          Mittens 
           
        </h3>
        <img
          src="https://i.insider.com/5cdeb255021b4c06ca38bbb3?width=2000&format=jpeg&auto=webp"
        />
      </a>
      <a
        class="doglink"
        href="/dogs/2"
      >
        <h3>
           
          Barky 
           
        </h3>
        <img
          src="https://cdn.kqed.org/wp-content/uploads/sites/39/2012/04/dog-nose-resize.jpg"
        />
      </a>
      <a
        class="doglink"
        href="/dogs/3"
      >
        <h3>
           
          Very Cool Guy
           
        </h3>
        <img
          src="https://ca-times.brightspotcdn.com/dims4/default/8139471/2147483647/strip/true/crop/2048x1365+0+0/resize/840x560!/format/webp/quality/90/?url=https%3A%2F%2Fcalifornia-times-brightspot.s3.amazonaws.com%2F1b%2F88%2F35dff51d0ac0da0d1f63ae4647a3%2Fla-tr-road-trip-issue-photos-20160621-010"
        />
      </a>
    </div>
  </div>
</div>
`;
